internal/meta/req.go

```diff
@@ -35,6 +37,9 @@ func FromTransportRequest(ctx context.Context, req *transport.Request) yarpc.Req
 // ToTransportRequest fills the given transport request with information from
 // the given ReqMeta. The Context associated with the ReqMeta is returned.
 func ToTransportRequest(reqMeta yarpc.CallReqMeta, req *transport.Request) context.Context {
+	if reqMeta == nil {
+		log.Panic("reqMeta must not be nil")
```

Comment:
I think it should be a panic because it's not really recoverable at runtime. There's no scenario under which you'll pass a nil ReqMeta, catch the error, and then pass a non-nil ReqMeta.